src/main/java/net/minecraftforge/common/loot/CanToolPerformAction.java

```diff
@@ -0,0 +1,61 @@
+package net.minecraftforge.common.loot;
+
+import com.google.common.collect.ImmutableSet;
+import com.google.gson.JsonDeserializationContext;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonSerializationContext;
+import net.minecraft.world.item.ItemStack;
+import net.minecraft.world.level.storage.loot.LootContext;
+import net.minecraft.world.level.storage.loot.parameters.LootContextParam;
+import net.minecraft.world.level.storage.loot.parameters.LootContextParams;
+import net.minecraft.world.level.storage.loot.predicates.LootItemCondition;
+import net.minecraft.world.level.storage.loot.predicates.LootItemConditionType;
+import net.minecraftforge.common.ToolAction;
+
+import javax.annotation.Nonnull;
+import java.util.Set;
+
+/**
+ * This LootItemCondition "forge:can_tool_perform_action" can be used to check if a tool can perform a given ToolAction.
+ */
+public class CanToolPerformAction implements LootItemCondition {
+
+    public static final LootItemConditionType LOOT_CONDITION_TYPE = new LootItemConditionType(new CanToolPerformAction.Serializer());
+
+    final ToolAction action;
+
+    public CanToolPerformAction(ToolAction action) {
+        this.action = action;
+    }
+
+    @Nonnull
+    public LootItemConditionType getType() {
+        return LOOT_CONDITION_TYPE;
+    }
+
+    @Nonnull
+    public Set<LootContextParam<?>> getReferencedContextParams() {
+        return ImmutableSet.of(LootContextParams.TOOL);
+    }
+
+    public boolean test(LootContext lootContext) {
+        ItemStack itemstack = lootContext.getParamOrNull(LootContextParams.TOOL);
+        return itemstack != null && itemstack.canPerformAction(this.action);
+    }
+
+    public static LootItemCondition.Builder canToolPerformAction(ToolAction action) {
+        return () -> new CanToolPerformAction(action);
+    }
+
+    public static class Serializer implements net.minecraft.world.level.storage.loot.Serializer<CanToolPerformAction> {
+        public void serialize(JsonObject json, CanToolPerformAction itemCondition, @Nonnull JsonSerializationContext context) {
+            json.addProperty("action", itemCondition.action.name());
+        }
+
+        @Nonnull
+        public CanToolPerformAction deserialize(JsonObject json, @Nonnull JsonDeserializationContext context) {
+            return new CanToolPerformAction(ToolAction.get(json.get("action").getAsString()));
```

Comment:
Is it worth having an JSON parsing error here for "ToolAction does not exist"? If the tool action is only "defined" via parsing the JSON, then the condition will never match, so an error might be nice to help with typos. I have implemented a similar system in one of my mods, in tool action terms it would work out to either `ToolAction.isDefined` or a nullable `ToolAction.getIfPresent`.

Only downside I can imagine to such a check is it prevents detecting a tool action that is dynamically defined after loot tables are parsed (e.g. defined via datapacks), but I cannot imagine why you would need that.

Reply:
I've already tried something like this. Unfortunately the ToolActions are not yet defined when the LootTables are loaded. So, I didn't include this here. 

I used the `Toolaction.getActions()` method, but the result was an empty list. :/

But an alternative could be that the name of the ToolAction instead of the ToolAction itself is saved in the LootContition. During the test method it could be compared against the existing ToolActions. If none exist, a warning can be logged.

Reply:
Aren't tool actions statically initialized? Means that they are not initialized because nothing referenced the `ToolActions` class. I suppose we could initialize it earlier, but there is no reason a mod would have to do so for its tools, meaning an error from undefined actions might cause more false positives than useful errors.

Reply:
Yes they are statically initialized. Maybe a registry could fix that? But including an extra registry just to display errors... Isn't that a bit exaggerated? ^^

Reply:
Making it a registry is not needed, just ensuring it's initialized before datapack load. That kinda violates the simple design of the tool actions as you now have to tell people when to get them

So, probably best to rely on datagen to minimize errors

Reply:
Do you have an idea how to ensure that it's initialized before datapack load?

Reply:
You can't, not really. It would require turning it into a registry-like setup with an event where mods are required to register the ToolActions. 
IMO, the best thing we can do is to just not have a warning. While it would be nice, it would make the system a lot more cumbersome.

Reply:
Ok, then I would just leave the part as it is.